### PR TITLE
gofumpt: add tests

### PR DESCRIPTION
**Pull request by legendofmiracles**
###### Motivation for this change
Followup to https://github.com/NixOS/nixpkgs/pull/154522

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by legendofmiracles** [comment]
Result of `nixpkgs-review pr 154583` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>gofumpt</li>
  </ul>
</details>

**Comment by legendofmiracles** [comment]
How can I tell editorconfig to ignore the failing file?

File: .editorconfig [hunk 1]
```diff
@@ -94,3 +94,6 @@ trim_trailing_whitespace = unset
 
 [pkgs/top-level/emscripten-packages.nix]
 trim_trailing_whitespace = unset
+
+[pkgs/development/tools/gofumpt/default.nix]
+trim_trailing_whitespace = unset
```
Comment (zowoq) [inline .editorconfig:99]: Please try this instead:

https://github.com/editorconfig-checker/editorconfig-checker#excluding-blocks

**Review by zowoq — CHANGES_REQUESTED** [review]

File: .editorconfig (hunk 1 above)
Comment (legendofmiracles) [reply .editorconfig]: Exactly what I was originally looking for! Thanks!

File: pkgs/development/tools/gofumpt/hello.go [hunk 2]
```diff
@@ -0,0 +1,7 @@
+package main
```
Comment (zowoq) [inline pkgs/development/tools/gofumpt/hello.go:1]: Does this ned to be checked in? Can it be inlined in the test?

Comment (legendofmiracles) [reply pkgs/development/tools/gofumpt/hello.go:1]: I have a strong hatred against multiline strings in shell, and to be honest I struggled a lot with the first, unformatted, heredoc... 

Now do the same but in a `test(1)` invocation - didn't feel like it. I've also never seen multiple line `test` code, it doesn't feel right :>

Comment (legendofmiracles) [reply pkgs/development/tools/gofumpt/hello.go:1]: If you want I can try to get it inlined - I do realize my reasons are purely selfish

Comment (zowoq) [reply pkgs/development/tools/gofumpt/hello.go:1]: I don't feel that strongly about it and I'm not going to insist on changes, I just dislike checking in non-essential code in separate language specific files.

TBH I don't think the package should have this type of test in tree, I'd expect the test will break before the package does. I'd just do a `testVersion` smoke check or similar and leave it at that.

Comment (Mic92) [reply pkgs/development/tools/gofumpt/hello.go:1]: I also think running tests from the upstream repo as part of our build should be enough.

Comment (legendofmiracles) [reply pkgs/development/tools/gofumpt/hello.go:1]: I figured it'd be less of a test but more of making the life of the reviewer easier. When I reviewed the update, I had to find some go repo (of course I couldn't think of one immediately, as it goes :)), clone it and then run it.

I don't think that the go formatting standards will be changing any time soon, especially with such simple code.

Comment (zowoq) [reply pkgs/development/tools/gofumpt/hello.go:1]: If you're keeping the test as it is please please add yourself as maintainer to the package.

**Pull request closed without merging**